From 284bd8093dfd717a4a7a9059ec00a9e395b5531f Mon Sep 17 00:00:00 2001
From: Suren Baghdasaryan <surenb@google.com>
Date: Fri, 28 Apr 2023 10:35:33 -0700
Subject: [PATCH 33/41] mm: change per-VMA lock statistics to be disabled by
 default

Change CONFIG_PER_VMA_LOCK_STATS to be disabled by default, as most
users don't need it. Add configuration help to clarify its usage.

Fixes: 52f238653e45 ("mm: introduce per-VMA lock statistics")
Suggested-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Suren Baghdasaryan <surenb@google.com>
Reviewed-by: Lorenzo Stoakes <lstoakes@gmail.com>
---
 mm/Kconfig.debug | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/mm/Kconfig.debug b/mm/Kconfig.debug
index f4b04d1c9..24df3a9a4 100644
--- a/mm/Kconfig.debug
+++ b/mm/Kconfig.debug
@@ -283,6 +283,12 @@ config DEBUG_KMEMLEAK_AUTO_SCAN
 config PER_VMA_LOCK_STATS
 	bool "Statistics for per-vma locks"
 	depends on PER_VMA_LOCK
-	default y
 	help
-	  Statistics for per-vma locks.
+	  Say Y here to enable success, retry and failure counters of page
+	  faults handled under protection of per-vma locks. When enabled, the
+	  counters are exposed in /proc/vmstat. This information is useful for
+	  kernel developers to evaluate effectiveness of per-vma locks and to
+	  identify pathological cases. Counting these events introduces a small
+	  overhead in the page fault path.
+
+	  If in doubt, say N.
-- 
2.41.0

